### PR TITLE
feat: copy over affilaites coingecko, add current price call

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David Adams <david@umaproject.org>

**Motivation**

https://app.clubhouse.io/uma-project/story/1220/add-coingecko-price-call
**Summary**

This is part of a [larger task to allow for querying collateral prices from api.](https://app.clubhouse.io/uma-project/story/930/create-api-to-get-current-prices-of-collateral-s)  Since collateral prices are typically easy to find, i will use coingecko to query prices rather that bot price feeds (which will be used later for synthetic prices).

Trying to break pr into smaller chunks to make easier to review. 

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

https://app.clubhouse.io/uma-project/story/1220/add-coingecko-price-call

### CODE REVIEW

File: packages/sdk/src/libs/coingecko/coingecko.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    this.host = host;
+  }
+  // Fetch historic prices for a `contract` denominated in `currency` between timestamp `from` and `to`. Note timestamps
+  // are assumed to be js timestamps and are converted to unixtimestamps by dividing by 1000.
+  async getHistoricContractPrices(contract: string, from: number, to: number, currency = "usd") {
+    assert(contract, "requires contract address");
+    assert(currency, "requires currency symbol");
+    assert(from, "requires from timestamp");
+    assert(to, "requires to timestamp");
+    from = from / 1000;
```
Comment (nicholaspai) [inline packages/sdk/src/libs/coingecko/coingecko.ts:17]: Do you need to `Math.round/floor/ceil` these at all?

File: packages/sdk/src/libs/coingecko/coingecko.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    to = to / 1000;
+    const { host } = this;
+    const result = await Coingecko.call(
+      `${host}/api/v3/coins/ethereum/contract/${contract.toLowerCase()}/market_chart/range/?vs_currency=${currency}&from=${from}&to=${to}`
```
Comment (nicholaspai) [inline packages/sdk/src/libs/coingecko/coingecko.ts:21]: nit: maybe add `/api/v3` to the `host` state var

**Review by nicholaspai — APPROVED** [review]
Just left some nit comments. Was really easy to review this as the PR is so small! Thanks

File: packages/sdk/src/libs/coingecko/coingecko.ts (hunk 1 above)
Comment (daywiss) [reply packages/sdk/src/libs/coingecko/coingecko.ts:17]: good call

**Comment by daywiss** [comment]
thanks for fast review @nicholaspai gonna merge this so i can unblock another pr

**Pull request merged**